packages/block-library/src/navigation-link/edit.js

```diff
@@ -200,7 +200,8 @@ export const updateNavigationLinkBlockAttributes = (
 	const kind = isCustomLink ? 'custom' : newKind;
 
 	setAttributes( {
-		...( url && { url: encodeURI( url ) } ),
+		// To prevent double application of encodeURI, decodeURI is executed to revert to the original string.
```

Comment:
Could you add a note on _when_ it might be possible to end up with double application? I assume it's if someone passes a URL that is already encoded?

Reply:
@getdave 

This is the case when a URL that has already been encoded is passed.
The URL suggested by LinkControl is retrieved using the WP-API. And the output of wp-api is URL encoded.

Reply:
Thanks for explaining. I've made a suggested change to code comment below. A developer coming to this may not have the context of this PR and so I'd favour being explicit about the reason for the change.

It's minor but I feel it's worth it.